File: jv-service/src/main/java/com/lcaohoanq/jvservice/domain/auth/IAuthService.java
package com.lcaohoanq.jvservice.domain.auth;

import com.lcaohoanq.jvservice.domain.auth.AuthPort.AccountRegisterDTO;
import com.lcaohoanq.jvservice.domain.token.TokenPort.RefreshTokenDTO;
import com.lcaohoanq.jvservice.domain.user.User;
import com.lcaohoanq.jvservice.domain.user.UserResponse;
import jakarta.mail.MessagingException;

public interface IAuthService {

    User register(AccountRegisterDTO accountRegisterDTO) throws Exception;
    AuthPort.LoginResponse login(String email, String password) throws Exception;
    UserResponse getUserDetailsFromToken(String token) throws Exception;
    void logout(String token, User user) throws Exception;
    void verifyOtpToVerifyUser(Long userId, String otp) throws Exception;
    void verifyOtpIsCorrect(Long userId, String otp) throws Exception;
    AuthPort.LoginResponse refreshToken(RefreshTokenDTO refreshTokenDTO) throws Exception;
    void sendEmailOtp(User existingUser) throws MessagingException;
}

File: server/shoppe/kt-service/src/main/kotlin/com/lcaohoanq/ktservice/domain/auth/IAuthService.kt
package com.lcaohoanq.ktservice.domain.auth

import com.lcaohoanq.ktservice.domain.token.TokenPort
import com.lcaohoanq.ktservice.domain.user.User
import com.lcaohoanq.ktservice.domain.user.UserPort

interface IAuthService {

    fun login(account: AuthPort.AuthRequest): AuthPort.AuthResponse
    fun register(newAccount: AuthPort.SignUpReq)
    fun getUserDetailsFromToken(token: String): UserPort.UserResponse
    fun getCurrentAuthenticatedUser(): User
    fun refreshToken(refreshTokenDTO: TokenPort.RefreshTokenDTO): AuthPort.AuthResponse
    fun logout(token: String, user: User)
}